go.mod

```diff
@@ -0,0 +1,51 @@
+module github.com/tendermint/tendermint
+
+go 1.12
+
+require (
+	github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
```

Comment:
can we structure this like Gopkg.toml?

```
# Allow only patch releases for serialization libraries
# Allow only minor releases for other libraries
## Test libraries
## XXX
## YYY
```

Reply:
Related note to myself: double check that all dependencies from the latest develop survived the mode to go modules (e.g. can't find "github.com/etcd-io/bbolt" which was introduced after this PR was opened)

Reply:
@melekes looks like part of the tooling will rewrite the go.mod file to be alphabetically sorted...